### ISSUE TITLE
Window resizing takes too long/delayed on Linux

### ISSUE THREAD

**Issue by Koranir** [zed-industries/zed#12577]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

It seems that Zed queues a redraw and resize of the window for every window resize request it receives (instead of resizing only to the latest received size), which results in it taking a fair amount of time to resize windows.

Steps to reproduce:
1. Open Zed
2. Resize window

(ignore the noise)

https://github.com/zed-industries/zed/assets/101683475/ac248f6f-1f54-483e-acea-bb59ba8d091b

### Environment

```
Zed: v1.0.0 (Zed)
OS: Linux 1.0.0
Memory: 31.2 GiB
Architecture: x86_64
```

Running KDE Plasma 6.1 beta on Wayland with an NVIDIA GTX 1660 Super, but issue is present on stable Plasma 6 & Hyprland.

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

[Zed.log](https://github.com/user-attachments/files/15524799/Zed.log)

**Comment by orual** [comment]
Confirmed also on Plasma 6.05 on AMD. Assuming this is already known to the devs and in the queue, but the Zed window doesn't have a resize handle which I could find, I had to use Super->RMB or the window menu to resize.

**Comment by amtoaer** [comment]
I found this issue on Plasma Wayland, too. A possible workaround might be to [launch Zed in X11 mode](https://github.com/zed-industries/zed/blob/main/docs/src/development/linux.md#wayland--x11). I remember it worked but can't confirm, cause I'm not using Plasma anymore.

**Comment by Koranir** [comment]
> I found this issue on Plasma Wayland, too. A possible workaround might be to [launch Zed in X11 mode](https://github.com/zed-industries/zed/blob/main/docs/src/development/linux.md#wayland--x11). I remember it worked but can't confirm, cause I'm not using Plasma anymore.

Running Zed through `WAYLAND_DISPLAY="" zed` resolves this issue, but causes flickering problems (which might be due to the NVIDIA drivers not supporting explicit sync yet).

**Comment by someone13574** [comment]
This has been happening on my m1 Mac on Linux as well. Wayland only.

**Comment by someone13574** [comment]
> Running Zed through `WAYLAND_DISPLAY="" zed` resolves this issue, but causes flickering problems.

This is a known issue introduced when they added window transparency: https://github.com/zed-industries/zed/issues/11469

**Comment by cattokomo** [comment]
This bug is also on Plasma Wayland and X1 with `WAYLAND_DISPLAY=""` on Intel.

**Comment by mikayla-maki** [comment]
People on Discord have said that these issues have cleared up on the latest main. Let us know if things change :)

**Comment by roworu** [comment]
Hi, I recently installed Zed on Fedora 40 + proprietary Nvidia drivers , using curl method, and facing very same issue, attaching logs:
[Zed.log](https://github.com/user-attachments/files/16176343/Zed.log)
I can see, that when I try to resize a Zed window, it spams these messages in log, a lot:
```
2024-07-11T14:47:28+03:00 [INFO] Using surface present mode MAILBOX
2024-07-11T14:47:28+03:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T14:47:29+03:00 [INFO] Using surface present mode MAILBOX
2024-07-11T14:47:29+03:00 [WARN] Unable to forbid exclusive full screen
```
Zed version:
```
zed --version
Zed 0.143.6 – /home/roworu/.local/zed.app/libexec/zed-editor

```

**Comment by roworu** [comment]
I checked workaround from above, to start Zed in XWayland session, and its working better. 
Seems like that bug reintroduced. Also video attached.


https://github.com/user-attachments/assets/b5c62e4d-3756-4a48-b16a-7ce1e7c415fa


@mikayla-maki , sorry for ping, but may I ask, you to reopen that issue, please?

**Comment by phisch** [comment]
Yes, this should be re-opened! A very easy way to reproduce this is by installing and running the [niri compositor](https://github.com/YaLTeR/niri) (smithay based).

**Comment by Koranir** [comment]
On latest KDE, COSMIC, and Niri, I cannot reproduce this problem. The 'surface present mode' errors are likely from a different source and should be a separate issue, if they still apply now.

**Comment by roworu** [comment]
Hello,

The issue is still persist for me on latest KDE with latest Nvidia drivers, may be I'm doing something wrong?
Retested both Wayland and XWayland modes, nothing indicative shown in logs:
https://github.com/user-attachments/assets/b608d3d2-71a8-418f-8c00-91bb56688c25


Some system info:
```
### Zed installation (used official install script from zed.dev):
zed --version
Zed 0.195.5 72015b2f6636e154680a570f402736b6c928a14f  – /home/roworu/.local/zed.app/libexec/zed-editor


### KDE `About page` data
Operating System: Fedora Linux 42
KDE Plasma Version: 6.4.3
KDE Frameworks Version: 6.16.0
Qt Version: 6.9.1
Kernel Version: 6.15.6-200.fc42.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 8 × AMD Ryzen 5 5600X 6-Core Processor
Memory: 32 GiB of RAM (31.3 GiB usable)
Graphics Processor: NVIDIA GeForce RTX 3070
```

Full log file: 

[Zed.log](https://github.com/user-attachments/files/21352893/Zed.log)

Can we reopen that one, please?
